pkg/controller/machine/machine.go

```diff
@@ -128,6 +128,10 @@ type NodeSettings struct {
 	HyperkubeImage string
 	// The kubelet repository to use. Currently only Flatcar Linux uses it.
 	KubeletRepository string
+	// Translates to feature gates on the kubelet.
+	// Expected input: FeatureGateName=true|false,...
+	// Default: RotateKubeletServerCertificate=true
+	KubeletFeatureGates []string
```

Comment:
Probably this need to be changed to `map[string]bool`, and `userdatahelper.KubeletFeatureGates()` used to init and mainly default this map.

Reply:
We'd have to move `userdatahelper.KubeletFeatureGates()` to `cmd/`, as this struct is populated when setting up a controller in the `cmd/machine-controller` package. That makes sense to me, so should I go ahead and change this?

Reply:
Or, we could have a "constructor" for `machinecontroller.NewNodeSettings(opts ...NodeSettingsOptions)` being called like 

```
machinecontroller.NewNodeSettings(machinecontroller.WithFeatureGates(kubeletFeatureGates...))
```

Where `WithFeatureGates()` is defined like `func WithFeatureGates(featureGates ...string) map[string]bool` and will default feature gates properly.

Reply:
Of course all this will be called in `cmd/machine-controller/main.go`

Reply:
I haven't implemented the constructor, but I moved the parsing and defaulting logic to `cmd/machine-controller/main.go`. PTAL and let me know what do you think about this solution.